nevergrad/optimization/base.py

```diff
@@ -98,6 +98,7 @@ def __init__(
         # you can also replace or reinitialize this random state
         self.num_workers = int(num_workers)
         self.budget = budget
+        self.restricted_budget = False  # False if we can extend the number of ask.
```

Comment:
isnt that only true for one shots? maybe we could merge this descriptions?
also, I would tend to make it a class attribute instead, so that we can identify which classes work or not without instantiating it, and that would mean adding it to ConfiguredOptimizer as well (and dealing with different cases there for DiscreteBSO

Reply:
Not all one shot actually.
Some of them, but not all.

There is a complicated issue as you point out with ConfiguredOptimizer, which is the reason for which I did not use a class attribute.

This will remain draft for some time --- other priorities for now.

Reply:
> There is a complicated issue as you point out with ConfiguredOptimizer, which is the reason for which I did not use a class attribute.

the reason is that we handle two different objects when dealing with optimizers: classes and Configured optimizer instances, so yes, this requires changes in two places in that case, but I think that is the way you would need to go if I get the aim of what you want to do, or else I'd need more inputs.

Another completely different option would be to make sure that restricted budgets optimizer revert to something else when they are over, but that can be messy as well :s